src/proof.rs

```diff
@@ -22,166 +22,357 @@ use poseidon252::{
+        // Get the corresponding `StorageBid` value that for the `Bid`
+        // which is effectively the value of the proven leaf (hash of the Bid)
+        // and allocate it.
+        let bid_hash =
+            AllocatedScalar::allocate(composer, StorageScalar::from(bid).0);
+        // Allocate Bid-internal fields
+        let bid_hashed_secret =
+            AllocatedScalar::allocate(composer, bid.hashed_secret);
+        let bid_cipher = (
+            composer.add_input(bid.encrypted_data.cipher()[0]),
+            composer.add_input(bid.encrypted_data.cipher()[1]),
+        );
+        let bid_commitment = Point::from_private_affine(composer, bid.c);
+        let bid_stealth_addr = (
+            Point::from_private_affine(
+                composer,
+                bid.stealth_address.pk_r().into(),
+            ),
+            Point::from_private_affine(
+                composer,
+                bid.stealth_address.R().into(),
+            ),
+        );
+        let bid_elegibility_ts =
+            AllocatedScalar::allocate(composer, bid.elegibility_ts);
+        let bid_expiration_ts =
+            AllocatedScalar::allocate(composer, bid.expiration_ts);
+        // Allocate bid-needed inputs
+        let secret_k = AllocatedScalar::allocate(composer, secret_k);
+        let seed = AllocatedScalar::allocate(composer, seed);
+        let latest_consensus_step =
+            AllocatedScalar::allocate(composer, latest_consensus_step);
+        let latest_consensus_round =
+            AllocatedScalar::allocate(composer, latest_consensus_round);
+        // Decrypt the cypher using the secret and allocate value & blinder
+        let decrypted_data = bid.encrypted_data.decrypt(
+            self.secret
+                .as_ref()
+                .ok_or_else(|| CircuitErrors::CircuitInputsNotFound)?,
+            &bid.nonce,
+        )?;
+        let bid_value = AllocatedScalar::allocate(composer, decrypted_data[0]);
+        let bid_blinder =
+            AllocatedScalar::allocate(composer, decrypted_data[1]);
+        // Allocate the bid tree root to be used later by the score_generation
+        // gadget.
+        let bid_tree_root = AllocatedScalar::allocate(composer, branch.root);
+
+        // ------------------------------------------------------- //
+        //                                                         //
+        //                     BlindBid Circuit                    //
+        //                                                         //
+        // ------------------------------------------------------- //
+
+        // 1. Merkle Opening
+        let root = merkle_opening_gadget(
+            composer,
+            branch.clone(),
+            bid_hash.var,
+            branch.root,
+        );
+        // Add PI constraint for the root to the PI constructor
+        pi.push(PublicInput::BlsScalar(
+            -branch.root,
+            composer.circuit_size(),
+        ));
```

Comment:
Is this supposed to be a double closure?

Reply:
WDYM by double closure? There's no closure in here. Is just pushing an enum to a vec.

Reply:
The format on my screen was squashed and messed up, so it placed the braces in the wrong place.